packages/language/src/language-server/cache/include-cache.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { LRUCache } from "lru-cache";
+import { TextDocument } from "vscode-languageserver-textdocument";
+
+const FIVE_MINUTES = 1000 * 60 * 5;
+const DEFAULT_LINE_CUTOFF = 100;
+const includeCache = new LRUCache({
+  max: 500,
+  ttl: FIVE_MINUTES,
+});
+
+export const includeCacheApi = {
+  get: includeCache.get.bind(includeCache),
+  set: includeCache.set.bind(includeCache),
+  has: includeCache.has.bind(includeCache),
+  clear: includeCache.clear.bind(includeCache),
+};
+
+export function getFileContentPreview(
+  key: string,
+  document: TextDocument,
+  lineCutoff = DEFAULT_LINE_CUTOFF,
+) {
+  const partialContent = includeCache.get(key);
+  if (partialContent) {
+    return partialContent.toString();
+  }
+  // load up the first (LINE_CUTOFF) lines of content from the file (semi-arbitrary cutoff)
+  const fileContent = document.getText({
+    start: { line: 0, character: 0 },
+    end: { line: lineCutoff, character: 0 },
+  });
+  const lineCount = fileContent.matchAll(/\n/g);
+  const partialContentText =
+    Array.from(lineCount).length > lineCutoff
+      ? fileContent + "\n...\n"
+      : fileContent;
```

Comment:
Suggestion: This is a rather inefficient way of testing the length of the `fileContent`:
```suggestion
  const partialContentText =
    document.getText().length > fileContent.length
      ? fileContent + "\n...\n"
      : fileContent;
```